src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -680,7 +680,9 @@ private enum ActivationInitializationStage
                                 SchedulingContext).Ignore();
                         }
 
-                        RerouteAllQueuedMessages(activation, null, "Failed InvokeActivate", exception);
+                        // Reject all of the messages queued for this activation.
+                        var activationFailedMsg = nameof(Grain.OnActivateAsync) + " failed";
+                        RejectAllQueuedMessages(activation, activationFailedMsg, exception);
```

Comment:
The activation is already marked as invalid and this is running inside of a lock on the activation. Is there room for a race?